### DISCUSSION TITLE
Передача данных в виджет

### DISCUSSION THREAD

**Discussion by promomasters** [cdek-it/widget#30] [Q&A] [answered]
Добрый день, 

версия виджета 3.8.

В нашем проекте (PHP) время от времени возникает сценарий, когда пользователь сформировал заказ, рассчитал доставку, а затем ушел со страницы (например, чтобы добавить еще позиций к заказу). После возвращения виджет загружается заново и теряет информацию о выбранном ПВЗ. Соответственно, пользователю приходится снова вызывать окно виджета и указывать ПВЗ или адрес доставки.

Можно ли как-то передать в виджет данные о ранее выбранном ПВЗ / Адресе доставки, чтобы по аналогии с методами resetParcels() и addParcels() пересчитать стоимость доставки, не прибегая к повторному выбору адреса доставки или ПВЗ?

**Comment by vermorag** [comment, marked as answer]
Добрый день. Это возможно. При инициализации можно передать объект `selected`: https://github.com/cdek-it/widget/blob/main/dist/cdek-widget.es.d.ts#L211 - это адреса пвз/двери.

**Reply by promomasters** [reply]
Можно пример, как будет выглядеть заполненный объект?

**Reply by vermorag** [reply]
Пример из админки вордпресса:
```
{
          apiKey: window.cdek.apiKey,
          sender: true,
          debug: true,
          requirePostcode: true,
          defaultLocation: 'Новосибирск',
          hideFilters: {
            type: true,
            have_cash: true,
            have_cashless: true,
            is_dressing_room: true
          },
          selected: {
            office: "NSK21",
            door: "Россия, Новосибирск, Большевистская улица, 101"
          }
}
```
